modules/statistics.py

```diff
@@ -60,6 +69,85 @@ def save(self, ranking: Optional[List[str]] = None) -> None:
             self.data.all_rankings.append(ranking)
         self._save_data()
 
+    @property
+    def all_players(self) -> Set[str]:
+        """Return the name of all players that have ever played."""
+        return {name for ranking in self.data.all_rankings for name in ranking}
+
+    def true_skill(self) -> Dict[str, Rating]:
+        """
+        Return TrueSkill Rating object for each player.
+
+        A TrueSkill rating is a normal distribution with a given mean and
+        variance. High means are indicative of good players, that way you
+        can compare players.
+        """
+        # First create rating objects for each player, assuming no prior skill
+        rankings = {
+            name: self.env.create_rating()
+            for name
+            in self.all_players
+        }
+
+        # Iterate over all the saved games we have in our catalogue
+        for game in self.data.all_rankings:
+            # We have a free-for-all, so each player are placed in their own
+            # teams.
+            rating_groups = [[rankings[name]] for name in game]
+
+            # The rank of the players are stored from best to worse.
+            ranks = list(range(len(game)))
+
+            # Update the rating for each player
+            ratings = self.env.rate(rating_groups=rating_groups, ranks=ranks)
+            for name, ranking in zip(game, ratings):
+                rankings[name] = ranking[0]
+
+        # Sort the dictionary by player rating before returning it
+        sorted_rankings = sorted(
+            rankings.items(),
+            key=lambda r: self.env.expose(r[1]),
+            reverse=True,
+        )
+        return dict(sorted_rankings)
```

Comment:
Why is the dictionary sorted to a list of tuples, and then returned as a dictionary again?

Reply:
Didn't know that from Python 3.6.1 dictionaries retain their order.